sql/src/main/java/io/crate/analyze/ShowStatementAnalyzer.java

```diff
@@ -77,6 +79,51 @@ AnalyzedStatement analyzeShowTransaction(Analysis analysis) {
         return newAnalysis.analyzedStatement();
     }
 
+    Query rewriteShowSessionParameter(ShowSessionParameter node) {
+        /*
+         * Rewrite
+         * <code>
+         *     SHOW { parameter_name | ALL }
+         * </code>
+         * To
+         * <code>
+         *     SELECT [ name, ] setting
+         *     FROM pg_catalog.pg_settings
+         *     [ WHERE name = parameter_name ]
+         * </code>
+         */
+        StringBuilder sb = new StringBuilder("SELECT ");
+        QualifiedName sessionSetting = node.parameter();
+        if (sessionSetting != null) {
+            sb.append("setting ");
+        } else {
+            sb.append("name, setting ");
+        }
+        sb.append("FROM pg_catalog.pg_settings ");
+        if (sessionSetting != null) {
+            sb.append("WHERE name = ");
+            singleQuote(sb, sessionSetting.toString());
+        }
+        return (Query) SqlParser.createStatement(sb.toString());
```

Comment:
I would also add an `order by` clause so that the outcome is always consistent.

Reply:
I guess the order is already deterministic as hit will hit a static table.

Reply:
ah, right